room/compiler-processing/src/main/java/androidx/room/compiler/processing/XElement.kt

```diff
@@ -75,3 +79,19 @@ fun XElement.isConstructor(): Boolean {
     }
     return this is XConstructorElement
 }
+
+/**
+ * Attempts to get a Javac [Element] representing the originating element for attribution
+ * when writing a file for incremental processing.
+ *
+ * In KSP a [KSFileAsOriginatingElement] will be returned, which is a synthetic javac element
+ * that allows us to pass originating elements to JavaPoet and KotlinPoet, and later extract
+ * the KSP file when writing with [XFiler].
+ */
+internal fun XElement.originatingElementForPoet(): Element? {
```

Comment:
oh was this always internal? i thought it was going to be public. anyways, i prefer internal unless we have to make it public (mostly because KSFileAsOriginatingElement is not a true implementation)

Reply:
you're right, it was public but I changed it back to internal since I had the same thought https://github.com/androidx/androidx/pull/155#discussion_r608116580